### PR TITLE
fix itests

### PR DESCRIPTION
**Pull request by chaudum**
* updated log4j2.properties so crate writes to correct log file
* updated es/upstream with fixes for
  * NaN/Infinity/-Infinity
  * parsing error_trace argument non-strict when building response
* enabled es-analysis-common module
* updated python doc tests

### CODE REVIEW

File: blackbox/docs/interfaces/http.txt [hunk 1]
```diff
@@ -287,7 +287,7 @@ To get more insight into what exactly went wrong an additional ``error_trace``
 GET parameter can be specified to return the stack trace::
 
     sh$ curl -sS -H 'Content-Type: application/json' \
-    ... -X POST '127.0.0.1:4200/_sql?pretty&error_trace=True' -d@- <<- EOF
+    ... -X POST '127.0.0.1:4200/_sql?pretty&error_trace=true' -d@- <<- EOF
```
Comment (mxm) [inline blackbox/docs/interfaces/http.txt]: Shouldn't be necessary as this is not case sensitive.

File: blackbox/docs/src/doc_tests/tests.py [hunk 2]
```diff
@@ -153,9 +153,9 @@ def tearDown(self):
     env={'JAVA_HOME': os.environ.get('JAVA_HOME', '')},
     settings={
          # The disk.watermark settings can be removed once crate-python > 0.21.1 has been released
-         "cluster.routing.allocation.disk.watermark.low" : "85%",
-         "cluster.routing.allocation.disk.watermark.high" : "90%",
-         "cluster.routing.allocation.disk.watermark.flood_stage" : "95%",
+        'cluster.routing.allocation.disk.watermark.low': '100k',
+        'cluster.routing.allocation.disk.watermark.high': '10k',
+        'cluster.routing.allocation.disk.watermark.flood_stage': '1k',
```
Comment (mxm) [inline blackbox/docs/src/doc_tests/tests.py]: I don't understand these limits. Why is high < low and won't we use more than 1kb of disk space in tests?

File: app/src/main/java/io/crate/bootstrap/CrateDB.java [hunk 3]
```diff
@@ -96,13 +96,15 @@ protected Environment createEnv(Terminal terminal, Map<String, String> settings)
...
+        // 2) Remove "path.conf" from command-line settings but use it as a conf path if exists
+        //    We need to remove it, because it was removed in ES6, but we want to keep the ability
+        //    to set it as CLI argument and keep backwards compatibility.
         String confPathCLI = settings.remove("path.conf");
         final Path confPath;
         if (confPathCLI != null) {
             confPath = Paths.get(confPathCLI);
         } else {
-            confPath = Paths.get(crateHomePath, "conf");
+            confPath = Paths.get(crateHomePath, "config");
```
Comment (mxm) [inline app/src/main/java/io/crate/bootstrap/CrateDB.java]: Good catch.

File: app/src/main/java/io/crate/bootstrap/CrateDB.java [hunk 4]
```diff
@@ -96,13 +96,15 @@ protected Environment createEnv(Terminal terminal, Map<String, String> settings)
             throw new IllegalArgumentException("Please set the environment variable CRATE_HOME or " +
                                                "use -Cpath.home on the command-line.");
         }
-        // 2) Remove path.conf from command-line settings but use it as a conf path if exists
+        // 2) Remove "path.conf" from command-line settings but use it as a conf path if exists
+        //    We need to remove it, because it was removed in ES6, but we want to keep the ability
+        //    to set it as CLI argument and keep backwards compatibility.
```
Comment (mxm) [inline app/src/main/java/io/crate/bootstrap/CrateDB.java]: +1 but could you leave the first line unchanged?

File: blackbox/docs/interfaces/http.txt (hunk 1 above)
Comment (chaudum) [reply blackbox/docs/interfaces/http.txt]: it is, because now the request would fail because of strict parsing of the GET parameter

**Pull request merged**